ecl/agentexec/agentexec.cpp

```diff
@@ -212,20 +221,25 @@ int CEclAgentExecutionServer::run()
 
 //---------------------------------------------------------------------------------
 
-typedef std::tuple<unsigned, const char *, const char *> ThreadCtx;
+typedef std::tuple<IJobQueueItem *, unsigned, const char *, const char *> ThreadCtx;
 
 // NB: WaitThread only used by if pool created (see CEclAgentExecutionServer ctor)
 class WaitThread : public CInterfaceOf<IPooledThread>
 {
+    CEclAgentExecutionServer &owner;
+    Owned<IJobQueueItem> item;
 public:
-    WaitThread(const char *_dali, const char *_apptype, const char *_queue) : dali(_dali), apptype(_apptype), queue(_queue)
+    WaitThread(CEclAgentExecutionServer &_owner, const char *_dali, const char *_apptype, const char *_queue)
+        : owner(_owner), dali(_dali), apptype(_apptype), queue(_queue)
     {
         isThorAgent = streq("thor", apptype);
     }
     virtual void init(void *param) override
     {
         auto &context = *static_cast<ThreadCtx *>(param);
-        std::tie(wfid, wuid, graphName) = context;
+        IJobQueueItem *_item;
+        std::tie(_item, wfid, wuid, graphName) = context;
```

Comment:
If item was a `Linked<IJobQueueItem>` could `item` be used in the tie as well (avoiding the local variable)?

Reply:
not solely.. the tuple is assigning by reference, for it to work the IJobQueueItem in the tuple needs to be a Linked as well

Not sure it's better, but I've changed it.

Reply:
I have never used tie, so I didn't know that.  wuid is a const char * in the tuple, but StringAttr in the class, so I didn't think it needed to be the same type, only assignable.